### PR TITLE
Add Lwd.map_and_set

### PR DESCRIPTION
**Pull request by OlivierNicole**
I found myself defining this function quite quickly when trying `Lwd`, so I think it could be a useful addition. I do not claim that `map_and_set` is necessarily the best name.

### CODE REVIEW

File: lib/lwd/lwd.mli [hunk 1]
```diff
@@ -67,6 +67,9 @@ val peek : 'a var -> 'a
 (** Observe the current value of the variable, without any dependency
     tracking. *)
 
+val map_and_set : ('a -> 'a) -> 'a var -> unit
```
Comment (Julow) [inline lib/lwd/lwd.mli:70]: For inspiration, the stdlib has `Map.update`.

Comment (OlivierNicole) [reply lib/lwd/lwd.mli]: That does read better.

**Comment by Julow** [comment]
This is also one of the first utility function I defined.

**Comment by let-def** [comment]
That looks good, thank you.
What do you think about a `may_update : ('a -> 'a option) -> 'a var -> unit` variant?

**Comment by Julow** [comment]
Instead, what do you think of not invalidating the document when the new value is the same as the old one ? (I also think that would be a useful property on `Lwt.t` values https://github.com/let-def/lwd/issues/38)

**Comment by let-def** [comment]
Thanks, I merged your PR and I will add a `may_update` function.

**Comment by OlivierNicole** [comment]
Thanks!

**Pull request merged**